panda/src/eagldisplay/PandaViewController.mm

```diff
@@ -0,0 +1,108 @@
+#import "PandaViewController.h"
+
+#include "Python.h"
+
+#include "EAGLGraphicsWindow.h"
+#include "graphicsEngine.h"
+#include "asyncTaskManager.h"
+#include "load_prc_file.h"
+#include "throw_event.h"
+
+@interface PandaViewController () {
+  NSThread *_pandaThread;
+  NSURL *_fullModulePath;
+}
+- (void)pythonModuleMain;
+- (void)queueForGraphicsWindowAttachmentWithCreation:(BOOL)createWindow;
+@end
+
+@implementation PandaViewController
+
+- (void)startPythonApp:(NSString *)modulePath {
+  [self startPythonApp:modulePath pythonRoot:@"Frameworks/python"];
+}
+
+- (void)startPythonApp:(NSString *)modulePath pythonRoot:(NSString *)pythonRoot {
+  nassertv(modulePath != nil);
+  nassertv(pythonRoot != nil);
+
+  NSURL *url = [NSBundle.mainBundle bundleURL];
+
+  const char *python_home = [NSURL URLWithString:pythonRoot relativeToURL:url].fileSystemRepresentation;
+  wchar_t *wpython_home = Py_DecodeLocale(python_home, NULL);
+  Py_SetPythonHome(wpython_home);
+
+  // iOS provides a specific directory for temp files.
+  NSString *tmp_path = [NSString stringWithFormat:@"TMP=%@", NSTemporaryDirectory(), nil];
+  putenv((char *)[tmp_path UTF8String]);
+
+  _fullModulePath = [NSURL URLWithString:modulePath relativeToURL:url];
+
+  [self startPandaWithThreadFunction:@selector(pythonModuleMain)];
+}
+
+- (void)startCPPApp:(void *)mainFunction {
+  nassertv(mainFunction != NULL);
+  // (not yet implemented)
+}
+
+- (void)startPandaWithThreadFunction:(SEL)mainSelector {
+  // When starting up the engine, a GraphicsWindow is created for us, so no need
+  // to do so here.
+  [self queueForGraphicsWindowAttachmentWithCreation:false];
+  
+  _pandaThread = [[NSThread alloc] initWithTarget:self selector:mainSelector object:nil];
+  [_pandaThread start];
+}
+
+- (void)queueForGraphicsWindowAttachmentWithCreation:(BOOL)createWindow {
+  EAGLGraphicsWindow::vc_lock.lock();
+  while (EAGLGraphicsWindow::next_view_controller != nil) {
+    EAGLGraphicsWindow::vc_condition.wait();
+  }
+  EAGLGraphicsWindow::next_view_controller = self;
+  EAGLGraphicsWindow::vc_lock.unlock();
+  if (createWindow) {
+    // unimplemented
+    abort();
+  }
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+  PyGILState_STATE gstate = PyGILState_Ensure();
+  Py_Finalize();
+}
+
+- (void)pythonModuleMain {
+  PT(Thread) panda_thread = Thread::bind_thread("ios_app", "ios_app");
+
+  // TODO: Load a .prc file from bundle resources instead of hardocoding this.
+  const char *model_path_config = [NSString stringWithFormat:@"model-path %@", [_fullModulePath.path stringByDeletingLastPathComponent]].UTF8String;
```

Comment:
Possibly this could be improved by making use of (a generalised) mechanism in `configPageManager.cxx` that we use for other deploy-ng applications.  It currently looks for a special symbol that indicates an offset into the executable where PRC data is stored (though it could also be looking in the bundle resources instead).